src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,100 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private MyNode<K, V>[] table;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new MyNode[DEFAULT_CAPACITY];
+        threshold = (int) (LOAD_FACTOR * table.length);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (threshold >= size) {
+            int indexBucket = hashCode(key, table.length);
+            if (table[indexBucket] == null) {
+                table[indexBucket] = new MyNode<>(key, value, null);
+            } else {
+                MyNode<K, V> oldBucket = table[indexBucket];
+                while (oldBucket != null) {
+                    if (Objects.equals(key, oldBucket.key)) {
+                        oldBucket.value = value;
+                        return;
+                    } else if (oldBucket.next == null) {
+                        oldBucket.next = new MyNode<>(key, value, null);
+                        break;
+                    }
+                    oldBucket = oldBucket.next;
+                }
+            }
+            size++;
+        } else {
+            resize();
+            put(key, value);
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int indexBucket = hashCode(key, table.length);
+
+        if (table[indexBucket] == null) {
+            return null;
+        }
+
+        if (Objects.equals(key, table[indexBucket].key)) {
+            return table[indexBucket].value;
```

Comment:
redaundent empty line 50 
let's make this code easiaer , you don't need this checks if you use while loop  ,because anyway if you do not find equals key , you should return null